pkg/joinserver/grpc_deviceregistry_test.go

```diff
@@ -1605,7 +1453,9 @@ func TestDeviceRegistryProvision(t *testing.T) {
 				return
 			}
 			a.So(devs, should.BeEmpty)
-		})
+		}) {
+			t.FailNow()
```

Comment:
Why?
IMO we should only fail the test when we can't continue anymore - here we can easily go to the next testcase, since they should be independent and in fact could be executed in parallel.
Are you looking for `go test -failfast` ?